### PR TITLE
[bosesoundtouch] Fix regression and add tests

### PR DESCRIPTION
**Pull request by lsiepel**
Signed-off-by: lsiepel <leosiepel@gmail.com>

Fixes: https://github.com/openhab/openhab-addons/issues/14059

Two bugs where introduced with https://github.com/openhab/openhab-addons/pull/13842 
1. Replacing the deprecated library with SAXParserFactory it looked like a drop-in replacement. But `setNamespaceAware(true)` was essential. :-/
2. By refactoring XMLResponseHandler` a state was not preserved while it had to.

Based on XML provided by [@paeddib ](https://github.com/paeddib) i could it some tests to provent such regressions in the future. One test case is not finished and might need some help on that. The XML-input is verified, but i cannot seem to get a hold on the preserved channels. 

This PR should be cherry picked to go into 3.4.x as the binding in 3.4.0 is not in a working state.

Jar for 3.4.x : https://1drv.ms/u/s!AnMcxmvEeupwjplD0WsAcS4oPCnagg?e=QDxiXp

### CODE REVIEW

File: bundles/org.openhab.binding.bosesoundtouch/src/test/java/org/openhab/binding/bosesoundtouch/internal/SoundTouch20Tests.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ *
+ */
+@ExtendWith(MockitoExtension.class)
+@NonNullByDefault
+public class SoundTouch20Tests {
+
+    private @Mock @NonNullByDefault({}) ThingHandlerCallback thingHandlerCallback;
+    private @NonNullByDefault({}) Thing soundTouchThing;
+    private @NonNullByDefault({}) BoseSoundTouchHandler thingHandler;
+    private @NonNullByDefault({}) XMLResponseProcessor processer;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.bosesoundtouch/src/test/java/org/openhab/binding/bosesoundtouch/internal/SoundTouch20Tests.java:55]: typo: processor

File: bundles/org.openhab.binding.bosesoundtouch/src/test/java/org/openhab/binding/bosesoundtouch/internal/SoundTouch20Tests.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    @BeforeEach
+    public void initialize() {
+        // arrange
+        Configuration config = new Configuration();
+        config.put(BoseSoundTouchConfiguration.MAC_ADDRESS, "B0D5CC1AAAA1");
+
+        soundTouchThing = ThingBuilder.create(BoseSoundTouchBindingConstants.BST_20_THING_TYPE_UID, thingUID)
+                .withConfiguration(config).withChannel(ChannelBuilder.create(volumeChannelUID, "Number").build())
+                .withConfiguration(config).withChannel(ChannelBuilder.create(presetChannelUID, "Number").build())
```
Comment (lolodomo) [inline bundles/org.openhab.binding.bosesoundtouch/src/test/java/org/openhab/binding/bosesoundtouch/internal/SoundTouch20Tests.java:70]: You do not need `.withConfiguration(config)` twice

File: bundles/org.openhab.binding.bosesoundtouch/src/test/java/org/openhab/binding/bosesoundtouch/internal/handler/InMemmoryContentStorage.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.bosesoundtouch.internal.ContentItem;
+import org.openhab.core.storage.Storage;
+
+/**
+ * @author Leo Siepel - Initial contribution
+ */
+@NonNullByDefault
+public class InMemmoryContentStorage implements Storage<ContentItem> {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.bosesoundtouch/src/test/java/org/openhab/binding/bosesoundtouch/internal/handler/InMemmoryContentStorage.java:29]: Strange that this class is defined twice as identical in two different packages ?

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.bosesoundtouch/src/test/java/org/openhab/binding/bosesoundtouch/internal/handler/InMemmoryContentStorage.java (hunk 3 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.bosesoundtouch/src/test/java/org/openhab/binding/bosesoundtouch/internal/handler/InMemmoryContentStorage.java:29]: yeah, strange. not intentionally. removed it.

**Comment by lolodomo** [comment]
@jlaur : please cherry-pick in branch 3.4.x.

**Pull request merged**